EssentialFeed/EssentialFeedTests/Image Comment/Image Comment API/RemoteImageCommentsLoaderTests.swift

```diff
@@ -0,0 +1,178 @@
+												   username: "user")
+		let (comment2, commentJSON2) = makeImageItem(id: UUID(),
+												   message: "another message",
+												   created_at: (Date(timeIntervalSince1970: 1577881882), "2020-01-01T12:31:22+00:00"),
+												   username: "another user")
+		
+		let validJSON = makeItemsJSON([commentJSON1, commentJSON2])
+		
+		statusCodes.enumerated().forEach { index, code in
+			expect(sut: sut, toCompleteWith: .success([comment1,comment2]), when: {
+				client.complete(withStatusCode: code, data: validJSON)
+			})
+		}
+	}
+	
+	func test_load_doesNotDeliverResultAfterCancellingTask() {
+		let (sut, client) = makeSUT()
+		let nonEmptyData = Data("non-empty data".utf8)
+		
+		var received = [ImageCommentsLoader.Result]()
+		let task = sut.load { received.append($0) }
+		task.cancel()
+		
+		client.complete(withStatusCode: 404, data: anyData())
+		client.complete(withStatusCode: 200, data: nonEmptyData)
+		client.complete(with: anyNSError())
+		
+		XCTAssertTrue(received.isEmpty, "Expected no received results after cancelling task")
+	}
+	
+	func test_load_shouldNotDeliverResultOnDeallocation() {
+		let url = URL(string: "https://a-url.com")!
+		let client = HTTPClientSpy()
+		var sut: RemoteImageCommentsLoader? = RemoteImageCommentsLoader(client: client, url: url)
+		
+		var capturedResults = [RemoteImageCommentsLoader.Result]()
+		_ = sut?.load { capturedResults.append($0) }
+		
+		sut = nil
+		client.complete(withStatusCode: 200, data: makeItemsJSON([]))
+		
+		XCTAssertTrue(capturedResults.isEmpty)
+	}
+	
+	//MARK: Helpers
+	
+	private func makeSUT(url: URL = URL(string: "https://a-url.com")!, file: StaticString = #file, line: UInt = #line) -> (sut: RemoteImageCommentsLoader, client: HTTPClientSpy){
+		let client = HTTPClientSpy()
+		let sut = RemoteImageCommentsLoader(client: client, url: url)
+		trackForMemoryLeaks(client, file: file, line: line)
+		trackForMemoryLeaks(sut, file: file, line: line)
+		return (sut,client)
+	}
+	
+	private func expect(sut: RemoteImageCommentsLoader, toCompleteWith expectedResult: RemoteImageCommentsLoader.Result, when action: () -> Void, file: StaticString = #file, line: UInt = #line) {
+		_ = sut.load() { recievedResult in
+			switch (recievedResult, expectedResult) {
+			case let (.failure(recievedError as RemoteImageCommentsLoader.Error),
+					  .failure(expectedError as RemoteImageCommentsLoader.Error)):
+				XCTAssertEqual(recievedError, expectedError, file: file, line: line)
+			case let (.success(recievedComments),
+					  .success(expectedComments)):
+				XCTAssertEqual(recievedComments, expectedComments, file: file, line: line)
+			default:
+				XCTFail("Expected result: \(expectedResult), got: \(recievedResult) instead", file: file, line: line)
+			}
+		}
+		
+		action()
+	}
+	
+	private func makeImageItem(id: UUID, message: String, created_at: (date: Date, iso8601String: String), username: String) -> (model: ImageComment, json: [String: Any]) {
```

Comment:
The convention is to use camel-case:

```suggestion
	private func makeImageItem(id: UUID, message: String, createdAt: (date: Date, iso8601String: String), username: String) -> (model: ImageComment, json: [String: Any]) {
```